lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -269,9 +289,9 @@ def main():
     new_path = module.params.get('new_path')
     dup_ok = module.params.get('dup_ok')
     if state == 'present' and not new_name and not new_path:
-        cert = module.params.get('cert')
-        key = module.params.get('key')
-        cert_chain = module.params.get('cert_chain')
+        cert, key, cert_chain = load_data(cert=module.params.get('cert'),
```

Comment:
No, because someone could fat-finger a path such as `secretsss/cert.pem` when really the directory is called `secrets`. Then the `isfile` would fail and the module would try to upload the *string of the broken path* `secretsss/cert.pem` as the actual cert body (which would fail)